plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/JdbcModule.java

```diff
@@ -66,11 +68,17 @@ public void configure(Binder binder)
         bindSessionPropertiesProvider(binder, JdbcMetadataSessionProperties.class);
 
         binder.bind(JdbcClient.class).to(CachingJdbcClient.class).in(Scopes.SINGLETON);
-        binder.bind(ConnectionFactory.class).to(Key.get(ConnectionFactory.class, StatsCollecting.class));
 
         newOptionalBinder(binder, Key.get(int.class, MaxDomainCompactionThreshold.class));
     }
 
+    @Provides
+    @Singleton
+    public ConnectionFactory createConnectionFactory(@StatsCollecting ConnectionFactory connectionFactory)
```

Comment:
Silly question: how does Guava handle this apparent circular dependency? (Honestly don't know!)

Reply:
Right, thanks. (That's why I wrote "apparent")